homeassistant/components/tado/sensor.py

```diff
@@ -68,19 +77,22 @@ async def async_setup_entry(
 
         entities.extend(
             [
-                TadoZoneSensor(
-                    tado, zone["name"], zone["id"], variable, zone["devices"][0]
-                )
+                TadoZoneSensor(tado, zone["name"], zone["id"], variable)
                 for variable in ZONE_SENSORS[zone_type]
             ]
         )
 
     # Create device sensors
     for device in devices:
+        if "batteryState" in device:
+            device_type = TYPE_BATTERY
+        else:
+            device_type = TYPE_POWER
```

Comment:
Please set appropriate device class if possible.

https://developers.home-assistant.io/docs/core/entity/sensor#available-device-classes

Reply:
This is a local variable to distinguish between battery-powered devices and other devices, but it's not set publicly, so there's no need for setting the device class.

Reply:
What does a device with type power represent as measurement?